exec/java-exec/src/main/codegen/templates/DateIntervalFunctionTemplates/Extract.java

```diff
@@ -31,6 +31,48 @@
 
 public class ${className} {
 
+/* Dummy function template to allow Optiq to validate this function call.
+ * At DrillOptiq time we rewrite all date_part() functions to extract functions,
+ * since they are essentially the same
+ */
+@FunctionTemplate(name = "DATE_PART", scope = FunctionTemplate.FunctionScope.SIMPLE, nulls = FunctionTemplate.NullHandling.NULL_IF_NULL)
```

Comment:
I feel these new function implementations could be put into a separate patch, under a different JIRA, since they are different issue (although it's probably exposed by the function type change you made in this PR).